### PR TITLE
Add external binaries version to docker info

### PR DESCRIPTION
**Pull request by mlaventure**
Signed-off-by: Kenfe-Mickael Laventure <mickael.laventure@gmail.com>

--

This should prevent us to having to request that information on new issues.

### CODE REVIEW

**Comment by mlaventure** [comment]
@MHBauer 

It's mostly for users that are not using docker official packages. Or to spot easy to figure out "error" like them having replaced one of the binaries for testing and forgotten about it.

It's also nice to have it all in one place. It's just information.

File: hack/make/.go-autogen [hunk 1]
```diff
@@ -11,12 +13,16 @@ package dockerversion
 // Default build-time variable for library-import.
 // This file is overridden on build with build-time informations.
 const (
-	GitCommit string = "$GITCOMMIT"
-	Version   string = "$VERSION"
-	BuildTime string = "$BUILDTIME"
-	IAmStatic string = "${IAMSTATIC:-true}"
+	GitCommit          string = "7037123-unsupported"
```
Comment (LK4D4) [inline hack/make/.go-autogen]: I don't understand why this hardcoded

**Comment by LK4D4** [comment]
I think this should be solved by having `--version` flag for all binaries.

File: hack/make/.go-autogen (hunk 1 above)
Comment (mlaventure) [inline hack/make/.go-autogen]: Oups, bad commit -_-

**Comment by mlaventure** [comment]
@LK4D4 not all users have access to their daemon host

Also, I think having users only run 1 command to get this information is preferable than asking them to run `--version` on 3 of them.

**Comment by thaJeztah** [comment]
from a "support" perspective, I think it'd be great to have it all in `docker info`, but just my 0.02c  (haven't looked at the implementation)

**Comment by vdemeester** [comment]
I do agree with @thaJeztah and @mlaventure, having all in `docker info` makes sence (for remote and also because it's one and only command then 👼)

**Comment by justincormack** [comment]
Needs a rebase

**Comment by tonistiigi** [comment]
LGTM

File: daemon/daemon.go [hunk 2]
```diff
@@ -66,6 +66,9 @@ var (
 	// containerd if none is specified
 	DefaultRuntimeBinary = "docker-runc"
 
+	// DefaultInitBinary is the name of the default init binary
+	DefaultInitBinary = "docker-init"
```
Comment (jhowardmsft) [inline daemon/daemon.go]: Minor nit, but perhaps this should be in platform files with a comment that it's not used on Windows.

File: hack/make/.go-autogen [hunk 3]
```diff
@@ -11,12 +13,16 @@ package dockerversion
...
 const (
-	GitCommit string = "$GITCOMMIT"
-	Version   string = "$VERSION"
-	BuildTime string = "$BUILDTIME"
-	IAmStatic string = "${IAMSTATIC:-true}"
+	GitCommit          string = "$GITCOMMIT"
+	Version            string = "$VERSION"
+	BuildTime          string = "$BUILDTIME"
+	IAmStatic          string = "${IAMSTATIC:-true}"
+	ContainerdCommitID string = "${CONTAINERD_COMMIT}"
```
Comment (jhowardmsft) [inline hack/make/.go-autogen]: Minor nit too - perhaps if not on Windows.

**Review by lowenna — APPROVED** [review]
Minor nits which can be followed up later. Otherwise LGTM from the Windows side.

**Comment by vieux** [comment]
LGTM

**Comment by runcom** [comment]
This is missing `docker-proxy`

**Comment by mlaventure** [comment]
@jhowardmsft I'll make a PR to make those new fields unix only

@runcom it looks like docker-proxy doesn't have a `--version` or equivalent yet

**Pull request merged**